api/types/teleport_role.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// TeleportRole identifies the role of an SSH connection. Unlike "user roles"
+// introduced as part of RBAC in Teleport 1.4+ these are built-in roles used
+// for different Teleport components when connecting to each other.
+type TeleportRole string
```

Comment:
Any reason these aren't named just `Role` and `Roles` respectively?

Reply:
This would conflict with the RBAC `types.Role`, so I thought `TeleportRole` made sense. We could also name it SSHRole or something different.

Reply:
I'm not a huge fan of the name either, but can't think of anything better. @r0mant any ideas?

Reply:
Would `ClusterRole` be better? or maybe it's the "Role" part that is making it awkward?

Reply:
These are basically names of different Teleport components and services like auth server, proxy server, app server, etc., right? How about `SystemRole` or `ServiceRole`?